browser/js/common/directives/graphs/line-graph.directive.js

```diff
@@ -0,0 +1,115 @@
+app.directive('lineGraph', function(d3Service, $window) {
+    return {
+        restrict: 'EA',
+        scope: {
+            data: "=",
+            cols: "="
+        },
+        link: function(scope, ele, attrs) {
+
+            d3Service.d3().then(function(d3) {
+                let margin = { top: 20, right: 20, bottom: 30, left: 50 },
+                    width = 960 - margin.left - margin.right,
+                    height = 500 - margin.top - margin.bottom,
+                    dataColumns = scope.cols.split(","),
+                    dateTitles = ['year', 'date', 'time'],
+                    dateFormat;
+
+                let filteredData = scope.data.filter(obj => obj[dataColumns[0]] && obj[dataColumns[1]]);
+
+
+
+
+                //check if the data column header may contain date info ??
+                if (dataColumns[0] === 'year') {
```

Comment:
So only if the first selected column says `"year"`? This seems too specific, so I'd recommend you add something to your todo list so that you revisit this later—though it seems a reasonable assumption for the time being :)